In [2]:
import pandas as pd
import scipy
from scipy.stats import chi2_contingency


# PROBLEM STATMENT:
Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similar across regions.

In [3]:
df=pd.read_csv('E:\Data Science\Assignments\Hypothesis Testing\BuyerRatio.csv')

In [5]:
df

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

# H0 - All proportions of male-female buyer ratio is same
# HA - All proportions of male-female buyer ratio is not the same

In [6]:
df.drop('Observed Values',axis=1,inplace=True)

### Using chisquare test

In [7]:
scipy.stats.chi2_contingency(df)

Chi2ContingencyResult(statistic=1.595945538661058, pvalue=0.6603094907091882, dof=3, expected_freq=array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
       [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))


## P value is 0.66>0.05. Thus we accept H0.which means All proportions of male-female buyer ratio is same for all direction